File: study08/backend/csharp_company/Controllers/EmailController.cs
using Microsoft.AspNetCore.Mvc;
using company.Services;
using company.Models;

namespace company.Controllers
{
    [ApiController]
    [Route("company/[controller]")]
    public class EmailController : ControllerBase
    {
        private readonly EmailService _emailService;

        public EmailController(EmailService emailService)
        {
            _emailService = emailService;
        }

        [HttpPost("send")]
        public IActionResult SendEmail([FromBody] EmailRequest emailRequest)
        {
            if (emailRequest == null || string.IsNullOrEmpty(emailRequest.To) || string.IsNullOrEmpty(emailRequest.Subject) || string.IsNullOrEmpty(emailRequest.Body))
            {
                return BadRequest("Invalid email request.");
            }

            _emailService.SendEmail(emailRequest);
            return Ok("Email sent successfully.");
        }
    }
}

File: study08/backend/springframework5/company/src/main/java/com/company/controller/EmailController.java
package com.company.controller;

import com.company.service.EmailService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/email")
public class EmailController {

    private final EmailService emailService;

    @Autowired
    public EmailController(EmailService emailService) {
        this.emailService = emailService;
    }

    @PostMapping("/send")
    public ResponseEntity<String> sendEmail(@RequestParam String email,
                                            @RequestParam String subject,
                                            @RequestParam String message) {
        emailService.sendEmail(email, subject, message);
        return new ResponseEntity<>("Email sent successfully", HttpStatus.OK);
    }
}